ts/screens/messages/MessageRouterScreen.tsx

```diff
@@ -0,0 +1,135 @@
+import I18n from "../../i18n";
+import { mixpanelTrack } from "../../mixpanel";
+import { loadMessages } from "../../store/actions/messages";
+import { navigateToMessageDetailScreenAction } from "../../store/actions/navigation";
+import { messagesAllIdsSelector } from "../../store/reducers/entities/messages/messagesAllIds";
+import { messageStateByIdSelector } from "../../store/reducers/entities/messages/messagesById";
+import { GlobalState } from "../../store/reducers/types";
+import { InferNavigationParams } from "../../types/react";
+import { isStrictSome } from "../../utils/pot";
+import { MessageDetailScreen } from "./MessageDetailScreen";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  NavigationInjectedProps<InferNavigationParams<typeof MessageDetailScreen>>;
+
+/**
+ * In order to have the final CreatedMessageWithContentAndAttachments, these conditions should be verified:
+ * - props.allMessages should be pot.some
+ * - messageId should be in props.allMessages
+ * - props.messageState(messageId) should be defined and pot.some
+ * @param props
+ */
+const getLoadingState = (
+  props: Props
+): pot.Pot<
+  CreatedMessageWithContentAndAttachments | undefined,
+  string | undefined
+> => {
+  const messageId = props.navigation.getParam("messageId");
+  if (!isStrictSome(props.allMessages)) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "all Messages is not some"
+    });
+    return pot.map(props.allMessages, _ => undefined);
+  }
+  if (!props.allMessages.value.some(id => id === messageId)) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "id not in list"
+    });
+    return pot.noneError(`Cannot found ${messageId} in messages list`);
+  }
+  const messageState = props.messageState(messageId);
+  if (messageState === undefined) {
+    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
+      reason: "MessageState is undefined"
+    });
+    return pot.noneError("MessageState is undefined");
+  }
+  return messageState.message;
+};
+
+/**
+ * Choose the screen where to navigate, based on the message.content.eu_covid_cert
+ * @param message
+ * @param props
+ */
+const navigateToScreenHandler = (
+  message: CreatedMessageWithContentAndAttachments,
+  props: Props
+) => {
+  if (message.content.eu_covid_cert !== undefined) {
+    props.navigateToEuCovidCertificate(
+      message.content.eu_covid_cert.auth_code as EUCovidCertificateAuthCode,
+      message.id
+    );
+  } else {
+    props.navigateToDetails(message.id);
+  }
+};
+
+const MessageRouterScreen = (props: Props): React.ReactElement => {
+  // used to automatically dispatch loadMessages if the pot is not some at the first rendering
+  // (avoid displaying error at the first frame)
+  const [isFirstRendering, setFirstRendering] = useState(true);
+  const loadingState = getLoadingState(props);
+  const isLoading = !pot.isError(loadingState);
```

Comment:
```suggestion
  const isLoading = !pot.isError(loadingState) || pot.isLoading(props.allMessages);
```
without this condition you'll display an error for a while


https://user-images.githubusercontent.com/822471/120790236-f5ee3d80-c532-11eb-92ae-fecc9ca89fbb.mov

Reply:
What is it the configuration for this test?
This is my behaviour:

https://user-images.githubusercontent.com/26501317/120801303-4c627880-c541-11eb-82c9-5b67424c44f4.mov

This is strange because `loadingState` already includes `props.allMessages` pot state so that or condition shouldn't be necessary.

```typescript
 if (!isStrictSome(props.allMessages)) {
    void mixpanelTrack("MESSAGE_ROUTING_FAILURE", {
      reason: "all Messages is not some"
    });
    return pot.map(props.allMessages, _ => undefined);
  }
```
this blocks forwards the pot state for `props.allMessages`

Reply:
I done tests twice and now it works.
Maybe my setup was altered by my code editing for tests